src/par_iter/skip.rs

```diff
@@ -0,0 +1,78 @@
+use super::*;
+use super::internal::*;
+
+pub struct Skip<M> {
+    base: M,
+    n: usize,
+}
+
+impl<M> Skip<M> {
+    pub fn new(base: M, n: usize) -> Skip<M> {
+        Skip { base: base,
+               n: n }
+    }
+}
+
+impl<M> ParallelIterator for Skip<M>
+    where M: IndexedParallelIterator
+{
+    type Item = M::Item;
+
+    fn drive_unindexed<C>(self, consumer: C) -> C::Result
+        where C: UnindexedConsumer<Self::Item>
+    {
+        bridge(self, consumer)
+    }
+}
+
+impl<M> ExactParallelIterator for Skip<M>
+    where M: IndexedParallelIterator
+{
+    fn len(&mut self) -> usize {
+        let base_len = self.base.len();
+        if self.n > base_len {
```

Comment:
Hmm, I think I would not mutate `self.n` here, but instead just return `base_len.saturating_sub(self.n)`. It does seem a bit surprising to have `len()` mutate state. I should probably change it to `&self`; I think I just wanted to keep things maximally flexible in case computing the length required (e.g.) updating a cache or something. Not that I have an example where that would be the case.